src/frontend/src/lib/validation/principal.validation.ts

```diff
@@ -0,0 +1,16 @@
+import { Principal } from '@dfinity/principal';
+import { z } from 'zod';
+
+export const PrincipalTextSchema = z.string().refine(
+	(principal) => {
+		try {
+			Principal.fromText(principal);
+			return true;
+		} catch (err: unknown) {
+			return false;
+		}
+	},
+	{
+		message: 'Invalid textual representation of a Principal.'
```

Comment:
here I imagine no `i18n` because is dev error

Reply:
Because I copied the code 😉.

But yeah I would say it's a low level dev error, not sure we should add i18n there.